File: Ransom-Note.py
class Solution(object):
    def canConstruct(self, ransomNote, magazine):
        \\\
        :type ransomNote: str
        :type magazine: str
        :rtype: bool
        \\\
        count = [0] * 26
        for i in range(len(magazine)):
            count[ord(magazine[i]) - ord('a')] += 1
        for i in range(len(ransomNote)):
            idx = ord(ransomNote[i]) - ord('a')
            if count[idx] == 0:
                return False
            count[idx] -= 1
        return True

File: Ransom-Note.js
/**
 * @param {string} ransomNote
 * @param {string} magazine
 * @return {boolean}
 */
var canConstruct = function(ransomNote, magazine) 
{
    let count = new Array(26).fill(0);
    for (let i = 0; i < magazine.length; i++) 
    {
        count[magazine.charCodeAt(i) - 97]++;
    }
    for (let i = 0; i < ransomNote.length; i++) 
    {
        let idx = ransomNote.charCodeAt(i) - 97;
        if (count[idx] == 0) 
        {
            return false;
        }
        count[idx]--;
    }
    return true;
};
